lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -237,13 +243,25 @@ def log.trace?
         end
 
         if @watch
+          if @metadata_source
+            self.class.send(:include, KubernetesMetadata::WatchNamespace)
+            self.class.send(:include, KubernetesMetadata::WatchPod)
+          else
+            self.class.send(:include, KubernetesMetadata::WatchNamespaces)
+            self.class.send(:include, KubernetesMetadata::WatchPods)
+          end
+
           thread = Thread.new(self) { |this| this.start_pod_watch }
           thread.abort_on_exception = true
           namespace_thread = Thread.new(self) { |this| this.start_namespace_watch }
           namespace_thread.abort_on_exception = true
         end
       end
-      if @use_journal
+
+      if @metadata_source
+        log.debug "Will enrich record with metadata of #{@metadata_source.namespace_name}/#{@metadata_source.pod_name}/#{@metadata_source.container_name}"
```

Comment:
consider wrapping this in `log.on_debug do end` since you are building a string

Reply:
I tend to not build a string, but since you asked, I'm building one 😄